tests/beaker_tests/cisco_bgp_neighbor_af/test_bgpneighboraf.rb

```diff
@@ -492,36 +510,80 @@ def puppet_resource_cmd(af)
   get_namespace_cmd(agent, cmd, options)
 end
 
-# Search pattern for show run config testing
-def af_pattern(tests, id, af)
-  asn, vrf, nbr, afi, safi = af.values
-  if tests[id][:ensure] == :present
-    if vrf[/default/]
-      [/router bgp #{asn}/, /neighbor #{nbr}/,
-       /address-family #{afi} #{safi}/]
+def get_dependency_manifest(platform, af, remote)
+  extra_config = ''
+  if platform == 'ios_xr'
+    # XR requires the following before a vrf AF can be configured:
+    #   1. a global router_id
+    #   2. a global address family
+    #   3. route_distinguisher configured on the vrf
+    #   4. remote-as is required for neightbor
+    remote = 2 if remote.nil?
```

Comment:
Maybe better to add a default value in the function args?  
def get_dependency_manifest(platform, af, remote=2)

And can you rename ``remote`` to ``remote_as``?